File: recommender/models/movie.py
import enum
from db import db

from models.genre import GenreModel

class TypesEnum(enum.Enum):
    movie = 'movie'
    episode = 'episode'

genres = db.Table('movies_genres',
    db.Column('moviesId', db.Integer, db.ForeignKey('movies.id'), primary_key=True),
    db.Column('genresId', db.Integer, db.ForeignKey('genres.id'), primary_key=True)
)

class MovieModel(db.Model):
    __tablename__ = 'movies'

    id = db.Column(db.Integer, primary_key=True)
    imdbId = db.Column(db.String(100))
    title = db.Column(db.String(100))
    year = db.Column(db.Integer)
    rating = db.Column(db.String(5))
    releaseDate = db.Column(db.Date)
    genres = db.relationship('GenreModel', secondary=genres, lazy='subquery',
        backref=db.backref('movies', lazy=True))
    director = db.Column(db.String(100))
    # actors
    plot = db.Column(db.Text)
    # languages
    # countries
    poster = db.Column(db.String(100))
    type = db.Column(db.Enum(TypesEnum))
    production = db.Column(db.String(100))
    # ratings

    def __init__(self, imdbId, title, year, rating, release_date, director, plot, poster, type, production):
        self.imdbId = imdbId
        self.title = title
        self.year = year
        self.rating = rating
        self.releaseDate = release_date
        self.director = director
        self.plot = plot
        self.poster = poster
        self.type = type
        self.production = production

    @classmethod
    def find_by_id(cls, _id):
        return cls.query.filter_by(id=_id).first()

File: server/src/entities/Movie.ts
import { Entity, Column, PrimaryGeneratedColumn, ManyToMany, JoinTable, OneToMany } from 'typeorm';
import { Genre } from './Genre';
import { Actor } from './Actor';
import { Language } from './Language';
import { Country } from './Country';
import { Rating } from './Rating';
import { UserRating } from './UserRating';

enum movieTypes {
    movie,
    series,
    episode
}

@Entity({ name: 'movies' })
export class Movie {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({ length: 20 })
    imdbId: string;

    @Column({ length: 100 })
    title: string;

    @Column({ width: 4 })
    year: number;

    @Column({ length: 20 })
    rating: string;

    @Column('date')
    releaseDate: string;

    @ManyToMany(type => Genre, {
        cascade: true
    })
    @JoinTable({
        name: 'movies_genres',
        joinColumns: [
            {
                name: 'moviesId',
                referencedColumnName: 'id'
            }
        ],
        inverseJoinColumns: [
            {
                name: 'genresId',
                referencedColumnName: 'id'
            }
        ]
    })
    genres: Genre[];

    @Column({ length: 100 })
    director: string;

    @ManyToMany(type => Actor, {
        cascade: true
    })
    @JoinTable({
        name: 'movies_actors',
        joinColumns: [
            {
                name: 'moviesId',
                referencedColumnName: 'id'
            }
        ],
        inverseJoinColumns: [
            {
                name: 'actorsId',
                referencedColumnName: 'id'
            }
        ]
    })
    actors: Actor[];

    @Column('text')
    plot: string;

    @ManyToMany(type => Language, {
        cascade: true
    })
    @JoinTable({
        name: 'movies_languages',
        joinColumns: [
            {
                name: 'moviesId',
                referencedColumnName: 'id'
            }
        ],
        inverseJoinColumns: [
            {
                name: 'languagesId',
                referencedColumnName: 'id'
            }
        ]
    })
    languages: Language[];

    @ManyToMany(type => Country, {
        cascade: true
    })
    @JoinTable({
        name: 'movies_countries',
        joinColumns: [
            {
                name: 'moviesId',
                referencedColumnName: 'id'
            }
        ],
        inverseJoinColumns: [
            {
                name: 'countriesId',
                referencedColumnName: 'id'
            }
        ]
    })
    countries: Country[];

    @Column({ length: 255 })
    poster: string;

    @Column({ length: 255 })
    image: string;

    @Column('enum', { enum: movieTypes })
    type: string;

    @Column({ length: 100 })
    production: string;

    @OneToMany(type => Rating, rating => rating.movie, {
        cascade: true
    })
    ratings: Rating[];

    @OneToMany(type => UserRating, rating => rating.movie, {
        cascade: true
    })
    usersRatings: UserRating[];
}
